pkg/cmd/roachtest/main.go

```diff
@@ -319,6 +319,10 @@ func updateSpecForSelectiveTests(ctx context.Context, specs []registry.TestSpec)
 			specs[i].SkipDetails = "test skipped because it is stable and selective-tests is set."
 		} else {
 			selectedTestsCount++
+			if td, ok := tdMap[specs[i].Name]; ok {
+				// populate the stats as obtained from the test selector
+				specs[i].SetStats(td.AvgDurationInMillis, td.LastFailureIsPreempt)
```

Comment:
It seems like an unnecessary optimization. Why not set the stats for _all_ test specs, including those which are skipped? This way, we have fewer special cases; i.e., reduced complexity (owing to case analysis) is always a good trade-off unless an optimization is crucial.

Reply:
I will make the change